packages/api/internal/edge/edge.go

```diff
@@ -187,18 +188,21 @@ func NewCluster(ctx context.Context, endpoint string, endpointTls bool, secret s
 
 	endpointClient, err := api.NewClientWithResponses(endpointBaseUrl, clientAuthMiddleware)
 	if err != nil {
+		ctxCancel()
 		return nil, err
 	}
 
 	return &Cluster{
-		ctx: ctx,
+		ctx:       ctx,
+		ctxCancel: ctxCancel,
 
 		Id:     id,
 		Client: endpointClient,
 	}, nil
 }
 
 func (c *Cluster) Disconnect() {
+	c.ctxCancel()
 	<-c.ctx.Done()
```

Comment:
I think you don't have to wait for ctx.Done as it won't do anything (you cancel right before)